### PR TITLE
multi-env-map

### PR DESCRIPTION
**Pull request by spencercap**
added equirectangular support for skybox!

issue #256

### CODE REVIEW

File: src/misc/multi-env-map.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  if (!material) {
+    return [];
+  } else if (Array.isArray(material)) {
+    return material;
+  } else if (material.materials) {
+    return material.materials;
+  } else {
+    return [material];
+  }
+}
```
Comment (donmccurdy) [inline src/misc/multi-env-map.js:15]: I'm only aware of two possibilities here, and would expect this to be enough:

```js
return Array.isArray( material ) ? material : [ material ];
```

Do you know of situations where a mesh might not have a material, or where `material.materials` might come from?

File: src/misc/multi-env-map.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    });
+
+  });
+}
+
+/**
+ * Specifies an envMap on an entity, without replacing any existing material
+ * properties.
+ */
+AFRAME.registerComponent('multi-env-map', {
```
Comment (donmccurdy) [inline src/misc/multi-env-map.js:52]: let's just name this `env-map`, and then we can remove the other component. :)

File: src/misc/multi-env-map.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  multiple: true,
+
+  schema: {
+    path: {default: ''},
+    extension: {default: 'jpg', oneOf: ['jpg', 'png']},
+    type: {default: 'cube', oneOf: ['cube', 'equirectangular']},
+    fileName: {default: 'equirectangular'},
```
Comment (donmccurdy) [inline src/misc/multi-env-map.js:59]: If you don't mind, let's condense all of this into one property:

```
src: {default: []}
```

... where the cube type would require six (complete) URLs in set order, and the equirectangular type would require exactly one:

```html
<a-entity env-map="src: env/posx.png, env/negx.png, env/posy.png, ..." />
```

I think that will be simpler overall.

**Review by donmccurdy — COMMENTED** [review]
This looks great, thanks! Just a few comments. And, feel free to include an `@author` line at the top of the file if you'd like. :)

**Comment by dsinni** [comment]
Any word on the status of this?

**Comment by donmccurdy** [comment]
The feedback would need to be resolved before I can merge this, and three.js's environment setup has changed a bit in the past year too: the `scene.environment` is much easier to work with than assigning individual env maps to materials. I'm going to close this for now, but open to suggestions for addressing https://github.com/donmccurdy/aframe-extras/issues/256 in the future.

**Pull request closed without merging**